src/codegen/generators/statement_generator.rs

```diff
@@ -252,23 +252,31 @@ impl<'a, 'b> StatementCodeGenerator<'a, 'b> {
         statement: &AstControlStatement,
     ) -> Result<(), Diagnostic> {
         match statement {
-            AstControlStatement::If(ifstmt) => {
-                self.generate_if_statement(llvm_index, &ifstmt.blocks, &ifstmt.else_block)
-            }
+            AstControlStatement::If(ifstmt) => self.generate_if_statement(
+                llvm_index,
+                &ifstmt.blocks,
+                &ifstmt.else_block,
+                &ifstmt.end_location,
+            ),
             AstControlStatement::ForLoop(for_stmt) => self.generate_for_statement(
```

Comment:
we could also just pass the inner statements to the `generate_<control_statement_type>.statement` functions and let the functions themselves handle the extraction of the fields here, so we don't have to ignore the linter once again on the `generate_if_statement` implementation